### PR TITLE
#169 #167 Renamed InitMode.None to InitMode.Manual & Set Behavior default

### PR DESCRIPTION
**Pull request by AlexanderGipp**
InitMode.cs `None` => `Manual`

### CODE REVIEW

**Comment by AlexanderGipp** [comment]
(Sorry for not creating a feature branch, i probably should have done that)

**Comment by Keboo** [comment]
Fixes #169

File: AutoDI/SettingsAttribute.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@ namespace AutoDI
     [AttributeUsage(AttributeTargets.Assembly)]
     public class SettingsAttribute : Attribute
     {
-        public Behaviors Behavior { get; set; }
+        public Behaviors Behavior { get; set; } = Behaviors.Default;
```
[the remark this answers is not in the document]
Comment (AlexanderGipp) [reply AutoDI/SettingsAttribute.cs:8]: Eh i dont mind if those two changes are in one merge request, I can seperate them though if you want (Just let me know)

[the remark this answers is not in the document]
Comment (Keboo) [reply AutoDI/SettingsAttribute.cs:8]: Doesn't bother me. Thank you for the contribution!

**Comment by AlexanderGipp** [comment]
Oh boy

**Comment by Keboo** [comment]
What may not be obvious from the unit test failure is this library does some magic where it compiles a new assembly from a code snippet taken from within an existing C# file. So in particular, [this line](https://github.com/Keboo/AutoDI/blob/master/AutoDI.Build.Tests/ManualInjectionOfContainer.cs#L51) is what needs to update to use Manual rather than None.

**Comment by AlexanderGipp** [comment]
> 
> 
> What may not be obvious from the unit test failure is this library does some magic where it compiles a new assembly from a code snippet taken from within an existing C# file. So in particular, [this line](https://github.com/Keboo/AutoDI/blob/master/AutoDI.Build.Tests/ManualInjectionOfContainer.cs#L51) is what needs to update to use Manual rather than None.

Thanks i will take a look at that

**Comment by Keboo** [comment]
Sure happy to explain. 
First it is important to understand what this library does. When you use it, it add a step to the MSBuild pipeline that runs post-compile, that changes a bunch of the IL instructions to provide dependency injection. This is similar to how other projects like Fody or PostSharp work as well.

So in terms of writing tests, this project really needed something that would compile assemblies so that it could test the processing of assemblies and the resulting output. One option I considered was simple to build a bunch of test projects, but many of the tests I wanted to do only needed a small amount of code so I looked at developing something a bit easier to use that could read a C# code file and build an assembly from that. The result of that is the AutoDI.AssemblyGenerator project. 
This project is leveraging the Microsoft.CodeAnalysis nuget package that can be used to create a workspace and do compilations. To get its inputs its Execute method is leveraging the [CallerFilePath](https://docs.microsoft.com/dotnet/csharp/language-reference/attributes/caller-information) to automatically get the file path of the caller. So for the Generator class to work, its Execute method needs to be in the same physical file as the code that will be injected into the assembly. As a simplistic method for finding that code, it uses Regex to find the `<assembly />` comment, and simply slurps up everything to the corresponding `</assembly>`, using the other comments to specify things like assembly attributes and metadata. 
The last bit of the magic is the subtle switch that happens with the CreateInstance calls in the unit test. Using the test from the file linked above you will find `sut = _testAssembly.CreateInstance<Sut>();`. The issue is the generic parameter `Sut` is the wrong type in the unit test. Because when the unit test is compiled it is using the Sut class from the code that will be eventually compiled into the assembly generated by AutoDI.AssemblyGenerator. So the CreateInstance uses that type name to look in the generated assembly for a type that has the same namespace and type name, and returns an instance of that. This swapping can sometimes make some of the tests more complicated but servers the purpose of testing this library quite well.

**Comment by AlexanderGipp** [comment]
Thanks for the great explenation, I think you will need to add the `hacktoberfest-accepted` tag for the hacktoberfest thingy

**Pull request merged**